grype/presenter/sarif/presenter.go

```diff
@@ -0,0 +1,390 @@
+package sarif
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/owenrumney/go-sarif/sarif"
+
+	v3 "github.com/anchore/grype/grype/db/v3"
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/grype/internal/version"
+	"github.com/anchore/syft/syft/source"
+)
+
+// Presenter holds the data for generating a report and implements the presenter.Presenter interface
+type Presenter struct {
+	results          match.Matches
+	packages         []pkg.Package
+	srcMetadata      *source.Metadata
+	metadataProvider vulnerability.MetadataProvider
+}
+
+// NewPresenter is a *Presenter constructor
+func NewPresenter(results match.Matches, packages []pkg.Package, srcMetadata *source.Metadata, metadataProvider vulnerability.MetadataProvider) *Presenter {
+	return &Presenter{
+		results:          results,
+		packages:         packages,
+		metadataProvider: metadataProvider,
+		srcMetadata:      srcMetadata,
+	}
+}
+
+// Present creates a SARIF-based report
+func (pres *Presenter) Present(output io.Writer) error {
+	doc, err := pres.toSarifReport()
+	if err != nil {
+		return err
+	}
+	err = doc.PrettyWrite(output)
+	return err
+}
+
+// toSarifReport outputs a sarif report object
+func (pres *Presenter) toSarifReport() (*sarif.Report, error) {
+	doc, err := sarif.New(sarif.Version210)
+	if err != nil {
+		return nil, err
+	}
+
+	v := version.FromBuild().Version
+	if v == "[not provided]" {
+		// Need a semver to pass the MS SARIF validator
+		v = "0.0.0-dev"
```

Comment:
does version usually start with `v`? I can't remember (didn't check)

Reply:
good question. a `v#.#` would not be semver, so in fact this would be bad to return and i might need to check it

Reply:
Looks like it does not include a `v`, which is good. `grype -vv` outputs:
```
[0000]  INFO grype version: 0.33.1
```
... which is using the same `version.FromBuild().Version` effectively